plugins/woocommerce/includes/wc-rest-functions.php

```diff
@@ -373,3 +373,56 @@ function wc_rest_check_product_reviews_permissions( $context = 'read', $object_i
 function wc_rest_is_from_product_editor() {
 	return isset( $_SERVER['HTTP_X_WC_FROM_PRODUCT_EDITOR'] ) && '1' === $_SERVER['HTTP_X_WC_FROM_PRODUCT_EDITOR'];
 }
+
+/**
+ * Check if a REST namespace should be loaded. Useful to maintain site performance even when lots of REST namespaces are registered.
+ *
+ * @since 9.2.0.
+ *
+ * @param string $ns The namespace to check.
+ * @param string $rest_route (Optional) The REST route being checked.
+ *
+ * @return bool True if the namespace should be loaded, false otherwise.
+ */
+function wc_rest_should_load_namespace( string $ns, string $rest_route = '' ): bool {
```

Comment:
Could you add some tests for this function? It would be great to have some tests to ensure that the function works as expected. 🙏